app/src/store/data.ts

```diff
@@ -37,94 +48,288 @@ const lastBlockTimestamp = ref<number>(0);
-          matchingTokenContract.balanceOf(grantRoundAddress),
-        ]).then(
-          ([
-            // round details
-            startTime,
-            endTime,
-            metadataAdmin,
-            payoutAdmin,
-            registry,
-            metaPtr,
-            hasPaidOut,
-            // donation token details
-            donationTokenName,
-            donationTokenSymbol,
-            donationTokenDecimals,
-            // matching token details
-            matchingTokenName,
-            matchingTokenSymbol,
-            matchingTokenDecimals,
-            matchingTokenBalance,
-          ]) => {
-            // check for status against `now`
-            const now = Date.now();
+    const grantRoundsList = (await Promise.all(
+      roundAddresses.map(async (grantRoundAddress: string) => {
+        const data = await getGrantRound(grantRoundAddress);
+
+        return data?.grantRound;
+      })
+    )) as GrantRound[];
+
+    // Get latest set of contributions
+    const contributions = await fetchContributions(grantsDict, grantRoundsList[0].donationToken, forceRefresh);
+
+    // Pull all trust scores from gitcoin api
+    const trustBonusScores = await fetchTrustBonusScores(contributions.contributions, forceRefresh);
+
+    // Pull all contributions for the current GrantRounds
+    const grantRoundCLR: GrantRoundCLR[] = await Promise.all(
+      roundAddresses.map(async (grantRoundAddress: string, key: number) => {
+        const data = await fetchGrantRoundGrantData(
+          contributions.contributions,
+          trustBonusScores.trustBonus,
+          grantRoundAddress,
+          grantsDict,
+          grantRoundsList[key].matchingToken,
+          forceRefresh
+        );
+
+        return data.grantRoundCLR;
+      })
+    );
+
+    const grantDataByRound: { [key: string]: GrantRoundCLR } = grantRoundCLR.reduce(
+      (byRound: { [key: string]: GrantRoundCLR }, data: GrantRoundCLR): { [key: string]: GrantRoundCLR } => {
+        byRound[data.grantRound] = data;
+
+        return byRound;
+      },
+      {} as { [key: string]: GrantRoundCLR }
+    );
+
+    // Fetch Metadata
+    const grantMetaPtrs = grantsList.map((grant: Grant) => grant.metaPtr);
+    const grantRoundMetaPtrs = grantRoundsList.map((grantRound: GrantRound) => grantRound.metaPtr);
+    void (await fetchMetaPtrs(grantMetaPtrs, grantMetadata));
+    void (await fetchMetaPtrs(grantRoundMetaPtrs, grantRoundMetadata));
+
+    // Save off data
+    grants.value = grantsList as Grant[];
+    grantRounds.value = grantRoundsList as GrantRound[];
+    grantRoundsCLRData.value = grantDataByRound as { [key: string]: GrantRoundCLR };
+    grantContributions.value = contributions.contributions as { [hash: string]: Contribution };
+  }
+
+  async function syncWithLocalStorage(
+    key: string,
+    callback: (localStorageData?: LocalStorageData) => Promise<LocalStorageAnyObj>,
+    validityCheck?: (data: LocalStorageAnyObj) => boolean
+  ) {
```

Comment:
Can you add comments to all the methods here and document the function's purpose / input parameters / return data? For example it's not too clear what I should be doing with the callback and validity checks, and why this method even needs those, instead of syncing the full store each time

Reply:
I've tidied this up a little bit -- if we think this might add friction then I'm happy to refactor to use a setter/getter approach, I think this just read a little bit better?

Reply:
I've tidied this up a little bit -- if we think this might add friction then I'm happy to refactor to use a getter/setter approach, I think this just read a little bit better?